apis/k8ssandra/v1alpha1/k8ssandracluster_types_test.go

```diff
@@ -215,3 +216,25 @@ func TestGenerationChanged(t *testing.T) {
 	kc.ObjectMeta.Generation = 3
 	assert.True(kc.GenerationChanged())
 }
+
+func TestDcRemoved(t *testing.T) {
+	kcOld := createClusterObjWithCassandraConfig("testcluster", "testns")
+	kcNew := kcOld.DeepCopy()
+	kcOld.Spec.Cassandra.Datacenters = append(kcOld.Spec.Cassandra.Datacenters, CassandraDatacenterTemplate{
+		Meta: EmbeddedObjectMeta{
+			Name: "dc2",
+		},
+	})
+	require.True(t, DcRemoved(kcOld.Spec, kcNew.Spec))
```

Comment:
The following functionality (even if entirely broken) will pass this test:

```go
func DcRemoved(oldSpec K8ssandraClusterSpec, newSpec K8ssandraClusterSpec) bool {
  return true
}
```

This is a filler test which doesn't test any actual functionality. Least it could do is verify that before the append one would get false, like adding after line 222:

``require.False(t, DcRemoved(kcOld.Spec, kcNew.Spec))`` 

Or, to avoid faking that implementation with just:

```go
func DcRemoved(oldSpec K8ssandraClusterSpec, newSpec K8ssandraClusterSpec) bool {
  return len(oldSpec.Cassandra.Datacenters) == len(newSpec.Cassandra.Datacenters)
}
```

For example, just renaming some Datacenter name to verify that DcRemoved returned true even if the lengths match. And same sort for the DcAdded. 

Because right now it would be pretty easy to "refactor" and make tests pass while actually breaking the entire functionality and we do want to avoid that.

Reply:
I think you're taking this a bit too far - you can almost always write an implementation that will pass a test but still be wrong SOMEHOW. 

But I've done as you suggested even so.